homeassistant/components/mqtt/tag.py

```diff
@@ -127,13 +138,16 @@ async def subscribe_topics(self) -> None:
         """Subscribe to MQTT topics."""
 
         async def tag_scanned(msg: ReceiveMessage) -> None:
-            tag_id = self._value_template(msg.payload, "").strip()
-            if not tag_id:  # No output from template, ignore
+            tag_id: ReceivePayloadType = self._value_template(msg.payload, "").strip()
+            if not tag_id or isinstance(
+                tag_id, bytes
+            ):  # No output from template, ignore
```

Comment:
Let's keep this for a follow-up PR
```suggestion
            tag_id = self._value_template(msg.payload, "").strip()
            if not tag_id:  # No output from template, ignore
```